docs/scripts/render_hubs.py

```diff
@@ -0,0 +1,117 @@
+]
+
+hub_list = []
+for cluster_info in clusters:
+    cluster_path = cluster_info.parent
+    if "schema" in cluster_info.name or "staff" in cluster_info.name:
+        continue
+    # For each cluster, grab it's YAML w/ the config for each hub
+    yaml = cluster_info.read_text()
+    cluster = safe_load(yaml)
+
+    # Pull support chart information to populate fields (if it exists)
+    support_files = cluster.get("support", {}).get("helm_chart_values_files", None)
+
+    # Incase we don't find any Grafana config, use an empty string as default
+    grafana_url = ""
+
+    # Try to identify the data centre location of the cluster
+    # For kubeconfig and Azure providers, this will always default to None since
+    # we do not store that information in the cluster.yaml file
+    #
+    # First try "zone" for GCP clusters
+    datacentre_loc = cluster.get(cluster["provider"], {}).get("zone", None)
+    if datacentre_loc is None:
+        # Try "region" for AWS clusters
+        datacentre_loc = cluster.get(cluster["provider"], {}).get("region", None)
+
+    # Loop through support files, look for Grafana config, and grab the URL
+    if support_files is not None:
+        for support_file in support_files:
+            with open(cluster_path.joinpath(support_file)) as f:
+                support_config = safe_load(f)
+
+            grafana_url = (
+                support_config.get("grafana", {}).get("ingress", {}).get("hosts", "")
+            )
+            # If we find a grafana url, set it here and break the loop, we are done
+            if isinstance(grafana_url, list):
+                grafana_url = grafana_url[0]
+                grafana_url = f"[{grafana_url}](http://{grafana_url})"
+                break
+
+    # Define the cloud provider for this cluster which we'll insert into the table below
+    # For clusters where we know the provider but can't guess it from the YAML file,
+    # we define a few custom mappings.
+    custom_providers = {"utoronto": "azure"}
+    for clustername, provider in custom_providers.items():
+        if cluster["name"] == clustername:
+            provider = provider
+        else:
+            provider = cluster["provider"]
+
+    # For each hub in cluster, grab its metadata and add it to the list
+    for hub in cluster["hubs"]:
+        # Domain can be a list
+        if isinstance(hub["domain"], list):
+            hub["domain"] = hub["domain"][0]
+
+        hub_list.append(
+            {
+                # Fallback to name if display_name isn't available
+                "name": hub.get("display_name", "name"),
+                "domain": f"[{hub['domain']}](https://{hub['domain']})",
+                "id": hub["name"],
+                "hub_type": hub["helm_chart"],
+                "grafana": grafana_url,
+                "cluster": cluster["name"],
+                "provider": provider,
+                "data center location": datacentre_loc,  # Americanising for you ;)
+            }
+        )
```

Comment:
I'm a little bit confused about how these two loops work together? The `for`'s seem to be at the same indentation so it looks like you run one loop first, then the other? From what I can work out, the first loop sets `provider` to `azure` and then gets used in the next loop, but I think this would set all providers to `azure` which isn't what we want? I will check the PR preview but yeah... I'm confused...

Reply:
Oh ok, because the loop over the cluster's is waaaaay up the file, got it